### PR TITLE
Refactor location script for EXO PS

### PR DESCRIPTION
**Pull request by victor-onofrei**

### CODE REVIEW

File: Utilities.ps1 [hunk 1]
```diff
@@ -79,13 +79,33 @@ function Get-ExchangeObjectLocation {
...
-    if ($isLocal -and (-not $isRemote)) {
-        return [ExchangeObjectLocation]::exchangeOnPremises
-    } elseif ($isRemote) {
-        return [ExchangeObjectLocation]::exchangeOnline
+    $isMailUser = $exchangeObjectTypeDetails -eq 'MailUser'
+
+    $sessionScope = Get-PSSession |
+        Where-Object { $_.State -eq 'Opened' -and $_.ConfigurationName -eq 'Microsoft.Exchange' } |
+        Select-Object -ExpandProperty ComputerName
+    if ($sessionScope -eq 'outlook.office365.com') {
```
Comment (revolter) [inline Utilities.ps1:87]: Leave empty lines before and after control blocks

File: Utilities.ps1 [hunk 2]
```diff
@@ -79,13 +79,33 @@ function Get-ExchangeObjectLocation {
...
+        if ($isLocal) {
+            return [ExchangeObjectLocation]::exchangeOnline
+        } elseif ($isMailUser) {
+            # return [ExchangeObjectLocation]::notAvailable
```
Comment (revolter) [inline Utilities.ps1:91]: Remove this dead code

File: Utilities.ps1 [hunk 3]
```diff
@@ -79,13 +79,33 @@ function Get-ExchangeObjectLocation {
...
+            $errorMessage = -join (
+                'You ran the script from PowerShell connected to Exchange Online ',
+                "and recipient $ExchangeObject is of type $exchangeObjectTypeDetails ",
```
Comment (revolter) [inline Utilities.ps1:94]:
```suggestion
                "and recipient $exchangeObject is of type $exchangeObjectTypeDetails ",
```

**Review by revolter — CHANGES_REQUESTED** [review]

File: Utilities.ps1 (hunk 1 above)
Comment (victor-onofrei) [reply Utilities.ps1:88]: Done

File: Utilities.ps1 (hunk 3 above)
Comment (victor-onofrei) [reply Utilities.ps1]: Done

**Review by revolter — APPROVED** [review]

**Pull request merged**